src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3659,6 +3660,42 @@ int journal_file_open(
         return r;
 }
 
+int journal_file_parse_uid(const char *path, uid_t *ret_uid, bool parse_archive) {
+        _cleanup_free_ char *buf = NULL;
+        _cleanup_free_ char *p = NULL;
+        const char *a, *b, *at;
+        int r;
+
+        assert(path);
+        assert(ret_uid);
+
+        r = path_extract_filename(path, &p);
+        if (r < 0)
+                return r;
+        if (r == O_DIRECTORY)
+                return -EISDIR;
+
+        a = startswith(p, "user-");
+        if (!a)
+                return -EREMOTE;
+        b = endswith(p, ".journal");
```

Comment:
so this means corrupted files cannot be parsed with this (i.e. those with a `.journal~`). for a generic helper like this this appears to be quite a shorcoming.

Reply:
For now I'm no seeing in which case that would be helpful.

Also I actually don't remember why the helper allows archived journal files to be parsed since for now no caller need that too.